### ISSUE TITLE
Rework confirm registration endpoint to support upcoming integrations

### ISSUE THREAD

**Issue by euanmillar** [opencrvs/opencrvs-core#6954]
[MOSIP](https://mosip.io/) - Open Source Identity Platform

### 🔁 Current state

Currently the country config [event-registration handler](https://github.com/opencrvs/opencrvs-countryconfig/blob/8928662e1d7719d93b57bd628ea885dfc59023b3/src/api/event-registration/handler.ts#L38) calls `workflow` -package to add the necessary identifiers for a child (a national id / NIN for example). We can implement MOSIP integration here, waiting for the MOSIP to respond and then calling the `CONFIRM_REGISTRATION_URL`. 

When a person dies, core has hardcoded ways of passing information to MOSIP.

#### Problems with the approach

* **Security:** instead of calling workflow directly via `CONFIRM_REGISTRATION_URL`, it needs to go via `gateway`-service to allow one entrypoint only to core 
* **Error handling:** we need a new `rejectRegistration` endpoint if MOSIP cannot register a NID
* **Country-config upgrading ease:** country-config needs to have less code to make upgrades easier
* **Core separation of concerns:** core now has code and references hardcoded to MOSIP - we want core to be integration-agnostic

### 🚀 Proposal

> [High-level diagram of the proposed flow](https://www.tldraw.com/r/5Yoezw6yg7_TPiOXdkPDm?d=v98.-610.3901.2009.page)

In short; the country config event-registration handler (which calls reject or confirm) should be able to be commented out and 'hijacked' by an integration like MOSIP.

* Allow an HTTP request with a correct token to confirm or reject the registration via gateway
* Import `event-registration` -route from MOSIP repository. This route calls the [OpenCRVS-side MOSIP-mediator](https://github.com/opencrvs/opencrvs-core/issues/7913)

**Comment by naftis** [comment]
https://github.com/opencrvs/opencrvs-core/tree/mosip-integration-v1
https://github.com/opencrvs/opencrvs-countryconfig/tree/mosip-integration-v1
https://github.com/opencrvs/mosip

Placeholders & ideation saved in these branches and repository. The MOSIP one is only a mock-service at this point and the integration ones aren't related to MOSIP though the branches are named so.

**Comment by naftis** [comment]
Technical overview of our internal implementation https://www.tldraw.com/r/5Yoezw6yg7_TPiOXdkPDm?v=-156,-604,3899,2009&p=page
